x-pack/plugins/stack_alerts/common/build_sorted_events_query.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export interface BuildSortedEventsQuery {
+  aggregations?: unknown;
+  index: string[];
+  from: string;
+  to: string;
+  filter: unknown;
+  size: number;
+  sortOrder?: 'asc' | 'desc' | undefined;
+  searchAfterSortId: string | number | undefined;
+  timeField: string;
+}
```

Comment:
Could we use the Elasticsearch client typing instead of introducing our own? 🤔 
Would avoid duplication and would make it less likely we make mistakes (as we have in the past, which is why we've been moving over to using the shared types where possible in TM)

Reply:
When I started implementing, I was trying to see how much, if any of the security solutions library functions we could reuse from their Custom Query/Threshold implementation, so this function is based heavily on their `buildEventsSearchQuery`, but I will look into using the ES client typings

Reply:
@gmmorris Is this along the lines of what you were thinking? https://github.com/elastic/kibana/pull/88528/commits/468ee3c53718e4e84ffe5f2cb9d74ec661e81636

Reply:
Yup, precisely.
We'll have to move these over to the new client implementation we're working on [here](https://github.com/elastic/kibana/issues/88668) once that's made available.

I suspect search alert might be merged first, so that'll probably fall on *me* to fix, but using these in the first place should make that easier 😆